### PR TITLE
Laboratory work #1, Karnauhov Andrey - 21FPL1

### PR DESCRIPTION
**Pull request by AndreyKarnauhov**

### CODE REVIEW

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -66,7 +88,20 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(frequencies, dict) or frequencies == {} \
+            or not isinstance(top, int) or isinstance(top, bool) or top <= 0:
+        return None
+    top_list = []
+    for item in frequencies.items():
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:95]: you can split in the loop item to key, value It is a standard way

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -66,7 +88,20 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
...
+        if not isinstance(item[0], str) or not isinstance(item[1], (float, int)):
+            return None
+        top_list.append(item)
+    top_list.sort(reverse=True, key=lambda word_and_frequency: word_and_frequency[1])
+    top_list = top_list[:top]
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:100]: Why dot'n you return right this?

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -82,7 +117,17 @@ def calculate_tf(frequencies: dict[str, int]) -> Optional[dict[str, float]]:
...
     """
-    pass
+    if not isinstance(frequencies, dict):
+        return None
+    total_words = 0
+    tf_dict = {}
+    for token, frequency in frequencies.items():
+        if not isinstance(token, str):
+            return None
+        total_words += frequency
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:127]: try to use `sum` and do this step in one line

**Review by artyomtugaryov — COMMENTED** [review]
Good job! Let's pythonize the code

File: lab_1_keywords_tfidf/main.py (hunk 2 above)
Comment (AndreyKarnauhov) [reply lab_1_keywords_tfidf/main.py]: Fixed

File: lab_1_keywords_tfidf/main.py (hunk 3 above)
Comment (AndreyKarnauhov) [reply lab_1_keywords_tfidf/main.py]: Done

File: lab_1_keywords_tfidf/main.py (hunk 1 above)
Comment (AndreyKarnauhov) [reply lab_1_keywords_tfidf/main.py]: UPD. Now there is no cycle at all

File: lab_1_keywords_tfidf/main.py [hunk 4]
```diff
@@ -99,12 +132,12 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
-    pass
+    if not check(term_freq, dict, str, float, True) or not check(idf, dict, str, float):
+        return None
+    return {key: term_freq[key] * (idf[key] if key in idf else math.log(47)) for key in term_freq}
 
 
-def calculate_expected_frequency(
-    doc_freqs: dict[str, int], corpus_freqs: dict[str, int]
-) -> Optional[dict[str, float]]:
+def calculate_expected_frequency(doc_freqs: dict[str, int], corpus_freqs: dict[str, int]) -> Optional[dict[str, float]]:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:140]: revert

File: lab_1_keywords_tfidf/main.py [hunk 5]
```diff
@@ -2,7 +2,29 @@
 Lab 1
 Extract keywords based on frequency related metrics
 """
-from typing import Optional, Union
+from typing import Optional, Union, Any
+from string import punctuation
+from math import log
+
+
+def check(obj: Any, exp_type: Any, exp_cont: Any = None, exp_val: Any = None, not_empty: bool = False) -> bool:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:10]: bad name for this function. Check what? Also - bad names of agruments

File: lab_1_keywords_tfidf/start.py [hunk 6]
```diff
@@ -31,6 +34,28 @@
     with open(CORPUS_FREQ_PATH, 'r', encoding='utf-8') as file:
         corpus_freqs = json.load(file)
 
-    RESULT = None
+    clean_text, frequencies, tf_calculated, tfidf_calculated, expected_frequency, \
+        chi_values, significant_words, RESULT = [None for not_undefined in range(8)]
+
+    tokenized_text = clean_and_tokenize(target_text)
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/start.py:40]: Use CAPS to name the variable outside of functions

**Review by artyomtugaryov — APPROVED** [review]

File: lab_1_keywords_tfidf/main.py (hunk 5 above)
Comment (AndreyKarnauhov) [reply lab_1_keywords_tfidf/main.py]: Fixed

File: lab_1_keywords_tfidf/start.py (hunk 6 above)
Comment (AndreyKarnauhov) [reply lab_1_keywords_tfidf/start.py]: Done

**Pull request closed without merging**